opensearch-project/opensearch-benchmark-workloads
# Rename challenges to test_procedures

Signed-off-by: Ian Hoang <hoangia@amazon.com>

### Description
When users invoke an OpenSearch-Benchmark command that uses "so" workload, there is an error. This is because the final rendered template is not rendered properly. An example is provided below. Please note the test_procedures is empty. This is because the directory within the "SO" directory is named as challenges instead of test_procedures. 

```
Last login: Mon Dec 20 11:17:05 on ttys015
hoangia@3c22fbd0d988 ~ % cat /var/folders/yh/89c2pcg10szgzwc6qj2h04gst115h_/T/tmpfch55370.json
 
 
{
  "version": 2,
  "description": "Indexing benchmark using up to questions and answers from StackOverflow",
  "indices": [
    {
      "name": "so",
      "body": "index.json"
    }
  ],
  "corpora": [
    {
      "name": "so",
      "base-url": "https://opensearch-benchmark-workloads.s3.amazonaws.com/corpora/so",
      "documents": [
        {
          "source-file": "posts.json.bz2",
          "document-count": 36062278,
          "compressed-bytes": 9600716233,
          "uncompressed-bytes": 35564808298
        }
      ]
    }
  ],
  "operations": [
        {
      "name": "index-append",
      "operation-type": "bulk",
      "bulk-size": 5000,
      "ingest-percentage": 100
    }
    
 
  ],
  "test_procedures": [
    
  ]
}%                                                                              
```

Successful output in the console after changing the directory from challenges to test_procedures:
```
(.venv) hoangia@3c22fbd0d988 opensearch-benchmark % opensearch-benchmark info --workload so

   ____                  _____                      __       ____                  __                         __
  / __ \____  ___  ____ / ___/___  ____ ___________/ /_     / __ )___  ____  _____/ /_  ____ ___  ____ ______/ /__
 / / / / __ \/ _ \/ __ \\__ \/ _ \/ __ `/ ___/ ___/ __ \   / __  / _ \/ __ \/ ___/ __ \/ __ `__ \/ __ `/ ___/ //_/
/ /_/ / /_/ /  __/ / / /__/ /  __/ /_/ / /  / /__/ / / /  / /_/ /  __/ / / / /__/ / / / / / / / / /_/ / /  / ,<
\____/ .___/\___/_/ /_/____/\___/\__,_/_/   \___/_/ /_/  /_____/\___/_/ /_/\___/_/ /_/_/ /_/ /_/\__,_/_/  /_/|_|
    /_/

[WARNING] Local changes in [/Users/hoangia/.benchmark/benchmarks/workloads/default] prevent workloads update from remote. Please commit your changes.
Showing details for workload [so]:

* Description: Indexing benchmark using up to questions and answers from StackOverflow
* Documents: 36,062,278
* Compressed Size: 8.9 GB
* Uncompressed Size: 33.1 GB

====================================================
TestProcedure [append-no-conflicts] (run by default)
====================================================

Indexes the whole document corpus using OpenSearch default settings. We only adjust the number of replicas as we benchmark a single node cluster and Benchmark will only start the benchmark if the cluster turns green. Document ids are unique so all index operations are append only.

Schedule:
----------

1. delete-index
2. create-index
3. check-cluster-health
4. index-append (8 clients)
5. force-merge
6. wait-until-merges-finish

-------------------------------
[INFO] SUCCESS (took 2 seconds)
-------------------------------
(.venv) hoangia@3c22fbd0d988 opensearch-benchmark % 

```
### Issues Resolved
#22 
 
### Check List

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
